Documentation/coding-guidelines/project-guidelines.md

```diff
@@ -53,122 +109,59 @@ Project configurations for a pure IL library project which targets the defaults.
 ```
 Project configurations with a unique implementation for each OS
 ```xml
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'FreeBSD_Debug|AnyCPU'" />
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'FreeBSD_Release|AnyCPU'" />
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Linux_Debug|AnyCPU'" />
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Linux_Release|AnyCPU'" />
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'OSX_Debug|AnyCPU'" />
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'OSX_Release|AnyCPU'" />
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Windows_Debug|AnyCPU'" />
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'Windows_Release|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'FreeBSD-Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'FreeBSD-Release|AnyCPU'" />
```

Comment:
> I do not think we have any assembly today that has different build between Linux and FreeBSD.

Yes and no.  Most of our assemblies that diverge based on platform just have a Windows vs Unix implementation.  However, there are a few that do specialize further with a Windows vs Linux vs macOS implementation, and for those few, in general the Linux implementation will not work out of the box on FreeBSD, so at the moment, we end up using a "not supported" build that throws exceptions from the specific pieces that would need specialized implementations, e.g.
https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.Process/src/System.Diagnostics.Process.csproj#L353-L357